Datahub.Core/Components/NotificationsIcon.razor

```diff
@@ -0,0 +1,214 @@
+    private bool ShowPreviousPageButton => CurrentPage > 1;
+    private bool ShowNextPageButton => CurrentPage < MaxPage;
+
+    private void ToggleNotifications() => ShowNotifications = !ShowNotifications;
+
+    private bool UserLanguageIsFrench => CultureInfo.CurrentUICulture.TwoLetterISOLanguageName.ToLowerInvariant() == "fr";
+
+    private async Task ToggleUnread(SystemNotification notification)
+    {
+        await systemNotificationService.SetReadStatus(notification.Notification_ID, !notification.Read_FLAG);
+    }
+
+    private async Task GoToActionLink(SystemNotification notification)
+    {
+        var t = ToggleUnread(notification);
+        if (!string.IsNullOrEmpty(notification.ActionLink_URL))
+        {
+            navManager.NavigateTo(notification.ActionLink_URL);
+        }
+        await t;
+    }
+
+    private async ValueTask<DateTimeOffset> GetLocalTime(SystemNotification notification)
+    {
+        var timestampUtc = DateTime.SpecifyKind(notification.Generated_TS, DateTimeKind.Utc);
+        return await timezoneService.GetLocalDateTime(timestampUtc);
+    }
+
+    private string GetUnreadCssClass(SystemNotification notification)
+    {
+        return notification.Read_FLAG ? "far fa-envelope" : "fas fa-envelope-open";
+    }
+
+    private string GetNotificationCardCssClass(SystemNotification notification)
+    {
+        var unread = notification.Read_FLAG ? "" : "unread";
+        return $"notification-item {unread}";
+    }
+
+    private async Task RefreshNotifications()
+    {
+        IsLoading = true;
+        StateHasChanged();
+        TotalNotificationCount = await systemNotificationService.GetNotificationCountForUser(CurrentUserId, UnreadOnly);
+        
+        UnreadNotificationCount = UnreadOnly ? 
+            TotalNotificationCount : 
+            await systemNotificationService.GetNotificationCountForUser(CurrentUserId, true);
+        
+        Notifications = await systemNotificationService.GetNotificationsForUser(CurrentUserId, UnreadOnly, CurrentPage);
+        IsLoading = false;
+        StateHasChanged();
+    }
+
+    private async Task ToggleShowUnreadOnly()
+    {
+        UnreadOnly = !UnreadOnly;
+        CurrentPage = 0;
+        await RefreshNotifications();
+    }
+
+    private async Task FirstPage()
+    {
+        CurrentPage = 0;
+        await RefreshNotifications();
+    }
+
+    private async Task NextPage()
+    {
+        CurrentPage = Math.Min(CurrentPage + 1, MaxPage);
+        await RefreshNotifications();
+    }
+
+    private async Task PrevPage()
+    {
+        CurrentPage = Math.Max(CurrentPage - 1, 0);
+        await RefreshNotifications();
+    }
+
+    public async Task OnNotify(string userId)
+    {
+        if (userId == CurrentUserId)
+        {
+            await InvokeAsync(() => RefreshNotifications());
+        }
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        PageSize = systemNotificationService.GetNotificationPageSize();
```

Comment:
No chance this can return zero?

Reply:
It's hardcoded to 5 but can be made configurable in the future. I can put a safety in this method to give it a sensible default just in case.